warehouse/query-core/src/main/java/datawave/query/index/stats/IndexStatsClient.java

```diff
@@ -106,9 +106,10 @@ public Map<String,Double> getStat(Set<String> fields, Set<String> dataTypes, Sor
             if (fields.isEmpty()) {
                 scanner = con.createScanner(table, auths);
             } else {
-                BatchScanner bScanner = con.createBatchScanner(table, auths, fields.size());
-                bScanner.setRanges(buildRanges(fields));
-                scanner = bScanner;
+                try (BatchScanner bScanner = con.createBatchScanner(table, auths, fields.size())) {
```

Comment:
try-with-resources will close the scanner at the end of the try block, but in the code, the scanner is used outside the try block.

Reply:
refactored this - should be better.